pkg/sql/sem/builtins/pg_builtins.go

```diff
@@ -265,7 +265,7 @@ func makeTypeIOBuiltin(paramTypes tree.TypeList, returnType *types.T) builtinDef
 				return nil, errUnimplemented
 			},
 			Info:       notUsableInfo,
-			Volatility: volatility.Volatile,
+			Volatility: volatility.Immutable,
```

Comment:
Heads up - you might know this already, but it looks like most volatility levels for these IO builtins are immutable; so, this will certainly reduce noise, but not to 0. There are some exceptions (at least with postgres v16.0) that are stable

Here are some that I can catch, just for an example:
```
array_recv
record_recv 
date_out
timestamptz_out
```

Here is a query that I use to find out the volatility:

```
select pg_get_function_arg_default(p.oid, 1), proname, pronamespace, n.nspname, proargtypes, prorettype, provolatile, proleakproof from pg_proc p join pg_namespace n on p.pronamespace = n.oid where proname ilike '%recv';
```
(and likewise for other suffixes: send, out, in)


Otherwise, lgtm 🙇